lib/baseFactory.js

```diff
@@ -139,12 +139,17 @@ class BaseFactory {
                         `expected Array, got ${typeof data}`);
                 }
                 const result = [];
+                const extraConfig = {
+                    datastore: this.datastore
+                };
+
+                // Only add scmPlugin if it's not a secret model
+                if (!config.secret) {
+                    extraConfig.scmPlugin = this.scmPlugin;
```

Comment:
You probably don't actually need to do this. It shouldn't hurt anything to pass extra parameters to the model, even if their value is `undefined`. If you decide you DO have to do this, be consistent and do it in `get` and `create`, too.